### PR TITLE
Return errors instead of panicking 

### PR DESCRIPTION
**Pull request by nathan-artie**

### CODE REVIEW

File: clients/snowflake/writes.go [hunk 1]
```diff
@@ -14,7 +15,10 @@ func (s *Store) Append(tableData *optimization.TableData) error {
 		if i > 0 {
 			if IsAuthExpiredError(err) {
 				slog.Warn("Authentication has expired, will reload the Snowflake store and retry appending", slog.Any("err", err))
-				s.reestablishConnection()
+				if connErr := s.reestablishConnection(); connErr != nil {
+					// TODO: Remove this panic and return an error instead
+					logger.Panic("Failed to reestablish connection", slog.Any("err", connErr))
```
Comment (nathan-artie) [inline clients/snowflake/writes.go:20]: Moved the panics up the call stack to preserve existing functionality.

File: lib/destination/utils/load.go [hunk 2]
```diff
@@ -50,27 +47,35 @@ func DataWarehouse(cfg config.Config, store *db.Store) destination.DataWarehouse
...
-			logger.Panic("Failed to clean up redshift", slog.Any("err", err))
+			return nil, fmt.Errorf("failed to clean up Redshift: %w", err)
 		}
-		return s
+		return s, nil
 	}
 
-	logger.Panic("No valid output sources specified", slog.Any("source", cfg.Output))
-	return nil
+	return nil, fmt.Errorf("ivalid data warehouse output source specified: %q", cfg.Output)
```
Comment (Tang8330) [inline lib/destination/utils/load.go:80]: typo

File: lib/destination/utils/load.go [hunk 3]
```diff
@@ -50,27 +47,35 @@ func DataWarehouse(cfg config.Config, store *db.Store) destination.DataWarehouse
 		})
 		return snowflake.LoadSnowflake(cfg, &store)
 	case constants.Snowflake:
-		s := snowflake.LoadSnowflake(cfg, store)
+		s, err := snowflake.LoadSnowflake(cfg, store)
+		if err != nil {
+			return nil, err
+		}
 		if err := s.Sweep(); err != nil {
```
Comment (Tang8330) [inline lib/destination/utils/load.go:54]: Err has been defined on L50, don't need `:=`, can just use `=`

Comment (Tang8330) [reply lib/destination/utils/load.go:54]: And ditto for all the other sweeps below

File: lib/destination/utils/load.go [hunk 4]
```diff
@@ -13,31 +13,28 @@ import (
...
 		}
 
-		return store
+		return store, nil
 	}
 
-	logger.Panic("No valid output sources specified", slog.Any("source", cfg.Output))
-
-	return nil
+	return nil, fmt.Errorf("ivalid baseline output source specified: %q", cfg.Output)
```
Comment (Tang8330) [inline lib/destination/utils/load.go:34]: typo

**Review by Tang8330 — APPROVED** [review]

**Pull request merged**